### PR TITLE
Improve treatment of RFCavity endpoints

### PR DESCRIPTION
**Pull request by cemitch99**
Modified the treatment of RF cavity endpoints in the quantity `efieldint` returned by the function `RF_Efield` in RFCavity.H.

This is a correction to the RF cavity field implementation that avoid spurious energy kicks associated with evaluation of the function `RF_Efield` at data values beyond the specified cavity endpoints.  The indefinite integral of Ez should be a nonzero constant outside the field range of interest, to ensure continuity across the two endpoints.

This has a small effect on the reference particle energy, and no modification of the benchmark analysis script analysis_rfcavity.py is needed.

### CODE REVIEW

**Review by qianglbl — APPROVED** [review]

File: examples/distgen/README.rst [hunk 1]
```diff
@@ -0,0 +1,261 @@
+
+Generation of beam distributions
+=================================
```
Comment (ax3l) [inline examples/distgen/README.rst:3]:
```suggestion
Generation of Beam Distributions
================================
```

File: examples/distgen/README.rst [hunk 2]
```diff
@@ -0,0 +1,261 @@
+
```
Comment (ax3l) [inline examples/distgen/README.rst]: Should this file be part of the current PR? I think it only belongs in #583

File: src/particles/elements/RFCavity.H [hunk 3]
```diff
@@ -377,6 +377,22 @@ namespace RFCavityData
...
+               }
+            }
+            else
+            {
+               efieldint = zmid*0.5_prt*cos_data[0];;
+               for (int j=1; j < m_ncoef; ++j)
+               {
+                 efieldint = efieldint - zlen*sin_data[j]*cos(j*pi)/(j*2*pi);
+               }
+            }
```
Comment (ax3l) [inline src/particles/elements/RFCavity.H]: Adding an inline comment and avoiding another code branch:
```suggestion
            else  // endpoint of the RF, outsize zlen
            {
               efieldint = std::copysign(z, z)*zmid*0.5_prt*cos_data[0];;
               for (int j=1; j < m_ncoef; ++j)
               {
                 efieldint = efieldint - zlen*sin_data[j]*cos(j*pi)/(j*2*pi);
               }
            }
```
https://en.cppreference.com/w/cpp/numeric/math/copysign

File: examples/distgen/README.rst (hunk 2 above)
Comment (cemitch99) [reply examples/distgen/README.rst]: No, agreed.

**Review by ax3l — APPROVED** [review]
:+1:

**Pull request merged**